image/image.go

```diff
@@ -41,19 +42,33 @@ type DockerAuth struct {
 	Auth     string `json:"auth,omitempty"`
 }
 
+// ParsePullSecret parses a kubernetes secret containing image pull credentials. Return either a
+// valid parsed secret or an error. If the secret does not exist and empty DockerConfigJson ref
+// is returned.
+func ParsePullSecret(secretClient corev1.SecretInterface, name string) (*DockerConfigJson, error) {
+	secret, err := secretClient.Get(name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return &DockerConfigJson{Auths: map[string]DockerAuth{}}, nil
+		}
+		return nil, err
+	}
+
+	secretData := secret.Data[strings.ReplaceAll(string(secret.Type), "kubernetes.io/", ".")]
+
+	var config *DockerConfigJson
+	if err = json.Unmarshal(secretData, &config); err != nil {
```

Comment:
`kubernetes.io/dockerconfigjson` and `kubernetes.io/dockercfg` have different structure. One of them looks like `{"auths":{"example.com":...}}`, another one like `{"example.com":...}`

Reply:
That came through my mind. I have chosen not to change this to avoid breaking other stuff. In other words: the code you are referring to here is the very same code that was already there prior to my change. The other format was never supported.

Reply:
I have added support for the SecretTypeDockercfg type as well. PTAL.
/cc @kleesc would you mind checking it once more ?